### PR TITLE
Add email notifications

### PR DESCRIPTION
**Pull request by andrew-gardener**
- gulp prod will now send a copy of the minified css to the templates folder for use in html email notifications
- Added mailhog to docker-compose as a development catch all mail server
- Added flower to docker-compose for an easy UI to view the status of worker tasks
- added two jinja functions (in 'compair/__init__') that are used in html email templates
- Added (simple) user notification settings to user form

Related: #381

### CODE REVIEW

File: alembic/versions/852abaee3a25_add_email_notifications_to_user_table.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+down_revision = '0e88581a5e5b'
+
+from alembic import op
+import sqlalchemy as sa
+
+from compair.models import convention
+
+
+def upgrade():
+    op.add_column('user', sa.Column('email_notifications', sa.Boolean(name='email_notifications'),
```
Comment (xcompass) [inline alembic/versions/852abaee3a25_add_email_notifications_to_user_table.py]: maybe change this to emun. we could have more types of the notifications,e.g. disable, enable, digest, etc.

File: celery_worker.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import os
 from compair import create_app, celery
 
-app = create_app(skip_endpoints=True, skip_assets=True)
+app = create_app()
```
Comment (xcompass) [inline celery_worker.py]: why skip_endpoints and skip_assets are removed?

Comment (andrew-gardener) [inline celery_worker.py]: oh, this is a mistake on my part. I originally had the email content generated in the delayed jobs (which required them to have access to the endpoints for url_for) so I removed it but celery had many problems using templates without a request context so I switch to just delivering the emails as a delayed task.

File: alembic/versions/852abaee3a25_add_email_notifications_to_user_table.py (hunk 1 above)
Comment (andrew-gardener) [inline alembic/versions/852abaee3a25_add_email_notifications_to_user_table.py]: Done

File: celery_worker.py (hunk 2 above)
Comment (andrew-gardener) [inline celery_worker.py]: Fixed

**Comment by lenglund** [comment]
Need to do an action that triggers a notification and go to 8025. You can view what's being sent (text and HTML).

**Comment by lenglund** [comment]
Can we start the email (text or HTML) with an intro?
"You've received <a new help comment / new answer feedback> for <course> on the following assignment:"

Are we not able to extract the feedback/comment text for the text version?

**Comment by lenglund** [comment]
Also, can we surface the email enable/disable, so a user can switch this on the profile form and on the profile screen? (Alternatively, we should at least note the selected option on the profile screen, i.e., "Yes" or "No".)

email

**Comment by andrew-gardener** [comment]
>Are we not able to extract the feedback/comment text for the text version?

At the time I decided not to since we'd have to strip all the html (which would mess up the formatting). 

I could try replace some expected tags (ex: `</p>`) with line breaks but there is no guarantee that it will look right in text format unless I look into something more complicated.

**Comment by lenglund** [comment]
No worries, I think we can text feedback leave as-is for now.

**Comment by andrew-gardener** [comment]
@lenglund modified email text and added notification toggle to profile screen (has same permissions as editing user profiles so admins + course instructors can also edit it)

**Pull request merged**